src/components/autocompleter/autocompleter.test.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { createRef } from 'react';
+import {
+  render, screen,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import jestServer from '../../mocks/jest-server.js';
+import Autocompleter from './autocompleter.jsx';
+import mockHandlers from './mock-handlers.js';
+
+describe('src/components/autocompleter/autocompleter', () => {
+  beforeEach(() => {
+    jestServer.use(...mockHandlers());
+  });
+
+  const requiredProps = {
+    id: 'test-id',
+    label: 'Test label',
+    name: 'field-id',
+    apiEndpoint: '/models',
+  };
+
+  it('has defaults', () => {
+    const ref = createRef();
+    render(<Autocompleter {...requiredProps} ref={ref} />);
+    expect(document.body).toMatchSnapshot();
+    expect(ref.current).toMatchSnapshot();
+  });
+
+  describe('className API', () => {
+    it('prioritizes className prop', () => {
```

Comment:
info for myself - what does `prioritizes` mean in this context? That it uses the class name provided and not the default or something?

If that's the case, should we make sure a class name _isn't_ there? If that's _not_ the case, can this test be a bit clearer like `it('applies the classname to the select')` or something?

Reply:
IDK, I copy and pasted it 😞

Reply:
Yeah.... I blame it on staying agile while writing tests. 😅  

I do have plans for making these copy-pasta tests as a factory of sorts.

Then we can make a new component with tests:

```
describe('foo', () => {
  apiTests({
    className: ['unique-class', () => expect(container.firstChild).toHaveClass('unique-class'),
    ...,
    value: ...,
  });
  
  it('has some unique behavior', () => {
    ...normal looking tests...
  });
});
```

Reply:
@cbacon91 FWIW I think
> If that's the case, should we make sure a class name isn't there?

is sufficiently covered by the snapshot.

Reply:
Most of our simple prop test do not seem worth while to me.